File: batches_java/batch_06_loop_type_change/ref.java
public class SumList {
    public static int sumList(int[] nums) {
        int total = 0;
        int i = 0;
        while (i < nums.length) {
            total += nums[i];
            i++;
        }
        return total;
    }

    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 4, 5};
        System.out.println(sumList(nums));
    }
}

File: batches_cpp/batch_06_loop_type_change/ref.cpp
#include <iostream>
#include <vector>
using namespace std;

int sumList(const vector<int>& nums) {
    int total = 0, i = 0;
    while (i < (int)nums.size()) { total += nums[i]; i++; }
    return total;
}

int main() {
    vector<int> nums = {1, 2, 3, 4, 5};
    cout << sumList(nums) << endl;
    return 0;
}
